demos/device/src/room/MeetingManager.tsx

```diff
@@ -0,0 +1,146 @@
+import {
+  AudioVideoFacade,
+  AudioVideoObserver,
+  ConsoleLogger,
+  DefaultDeviceController,
+  DefaultMeetingSession,
+  LogLevel,
+  MeetingSessionConfiguration,
+  ScreenShareViewFacade,
+  ScreenObserver,
+} from '../../../../build/index';
+
+const BASE_URL = [
+  location.protocol,
+  '//',
+  location.host,
+  location.pathname.replace(/\/*$/, '/').replace('/device', ''),
+].join('');
+
+class MeetingManager {
+  private meetingSession: DefaultMeetingSession;
+  private audioVideo: AudioVideoFacade;
+  private screenShareView: ScreenShareViewFacade;
+  private title: string;
+  // private device: MediaTrackConstraints = {audio: true}
+
+  async initializeMeetingSession(configuration: MeetingSessionConfiguration): Promise<any> {
+    const logger = new ConsoleLogger('DEV-SDK', LogLevel.DEBUG);
+    const deviceController = new DefaultDeviceController(logger);
+    configuration.enableWebAudio = false;
+    this.meetingSession = new DefaultMeetingSession(configuration, logger, deviceController);
+    this.audioVideo = this.meetingSession.audioVideo;
+    this.screenShareView = this.meetingSession.screenShareView;
+
+    await this.setupAudioDevices();
+  }
+
+  async setupAudioDevices(): Promise<void> {
+    const audioOutput = await this.audioVideo.listAudioOutputDevices();
+    const defaultOutput = audioOutput[0] && audioOutput[0].deviceId;
+    await this.audioVideo.chooseAudioOutputDevice(defaultOutput);
+
+    const audioInput = await this.audioVideo.listAudioInputDevices();
+    const defaultInput = audioInput[0] && audioInput[0].deviceId;
+    await this.audioVideo.chooseAudioInputDevice(defaultInput);
+  }
+
+  registerScreenShareObservers(observer: ScreenObserver): void {
+    if (!this.screenShareView) {
+      console.log('ScreenView not initialize. Cannot add observer');
+      return;
+    }
+    this.screenShareView.registerObserver(observer);
+  }
+
+  addObserver(observer: AudioVideoObserver): void {
+    if (!this.audioVideo) {
+      console.error('AudioVideo not initialized. Cannot add observer');
+      return;
+    }
+    this.audioVideo.addObserver(observer);
+  }
+
+  removeObserver(observer: AudioVideoObserver): void {
+    if (!this.audioVideo) {
+      console.error('AudioVideo not initialized. Cannot remove observer');
+      return;
+    }
+
+    this.audioVideo.removeObserver(observer);
+  }
+
+  bindVideoTile(id: number, videoEl: HTMLVideoElement): void {
+    this.audioVideo.bindVideoElement(id, videoEl);
+  }
+
+  async startLocalVideo(): Promise<void> {
+    const videoInput = await this.audioVideo.listVideoInputDevices();
+    const defaultVideo = videoInput[0];
+    await this.audioVideo.chooseVideoInputDevice(defaultVideo);
+    this.audioVideo.startLocalVideoTile();
+  }
+
+  stopLocalVideo(): void {
+    this.audioVideo.stopLocalVideoTile();
+  }
+
+  async startViewingScreenShare(screenViewElement: HTMLDivElement): Promise<void> {
+    this.meetingSession.screenShareView
```

Comment:
amazon-chime-sdk-js@1.2.1 introduced content-based screen sharing. You can now share screen capture or any media file. Please read details in https://aws.github.io/amazon-chime-sdk-js/modules/contentshare.html

Example in meetingV2:
https://github.com/aws/amazon-chime-sdk-js/blob/master/demos/browser/app/meetingV2/meetingV2.ts

Reply:
Added TODO to migrate to new API.